OPNSense 21.7.8 & 22.1 support & WPA_supplicant configuration file

zombielinux opened the pull request:
Updated code for using a wpa_supplicant configuration file and support for OPNsense 21.7.3. 

Notable change is the addition of spoofing the $ONT_IF mac address BEFORE bringing the interface up.

zombielinux committed:
Create opnatt-supplicant.sh

Successful configuration file to be placed in /usr/local/etc/syshook.d/early/99-opnatt. To be used in conjunction with a /conf/pfatt/wpa/wpa_supplicant.conf file.

zombielinux committed:
Create wpa_supplicant.conf

Added wpa_supplicant configuration file. The bash parsing of the list in the original opnatt.sh was complicated and unnecessary.

zombielinux committed:
Update opnatt-supplicant.sh

Removed some additional hardcoded entries.

Hou-dev commented:
Thanks for this. I was thinking of implementing pfatt like this. Will test this later and report on this comment. Much more clearer with separate files for supplicant and bridge methods. Edit: There seems to be some issue with the latest firmware opnsense 22.1 with my intel and broadcom nic. Stuck on wpa authentication. Made sure to check all files and locations and they are correct.
![original_a9086da7-f6e2-44af-a781-3837bd64f106_PXL_20220206_084810558](https://user-images.githubusercontent.com/59487209/152865542-cdba1f69-254c-48d0-99f4-bc1ec290c2c0.jpg)

MrCaturdayNight commented:
Make sure you grab the supplicant branch if you’re going to try that method. 

On Sat, Feb 5, 2022, at 6:24 PM, Hou-dev wrote:
> 
> 
> Thanks for this. I was thinking of implementing pfatt like this. Will test this later and report on this comment. Much more clearer if be had separate files for supplicant and bridge methods.
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/MonkWho/pfatt/pull/60#issuecomment-1030713487>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AVE72M42G6C64KB5UWGABWDUZWWR3ANCNFSM5H2XOIWQ>.
> You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>
> 
>

owenthewizard commented:
+1 would love to see this merged.

zombielinux committed:
Update opnatt-supplicant.sh

Explicitly load ng_ether module to prevent wpa_supplicant from hanging later in the script.

MrCaturdayNight commented:
This current version worked for me on 21.7.8 as well. Thanks!

zombielinux committed:
Adding OPNSense 22.1 support

Added "-vlanhwtag -vlanhwfilter -vlanhwtso" to line 50 to enable supplicant auth on OPNSense 22.1 per https://github.com/MonkWho/pfatt/issues/65

johnburbridge commented:
@zombielinux confirmed this works on a fresh install of 22.1.2_2.  What's interesting is that I did not have to assign ngeth0 to WAN. Its working with the previously assign physical interface (in my case igb0, on an Intel I350-AM4).

Am in the middle of upgrading to 22.1.4 but don't expect any issues.

Thanks so much for this! Would love to buy you a coffee.

UPDATE after upgrading from 22.1.2_2 -> 22.1.4:
The physical IF was still able to acquire its ip address but there was no route out.  On a whim, rather than manually assigning the route (which might change without notice) I switched over and assigned WAN to ngeth0 which made things start working again. It's not entirely clear if this was due to the upgrade or something to do with surviving a reboot, but I can confirm everything works after assigning WAN to ngeth0 and it does survive reboots.
